haystack/preview/dataclasses/document.py

```diff
@@ -143,42 +137,32 @@ def __eq__(self, other):
 
     def __post_init__(self):
         """
-        Generate the ID based on the init parameters and make sure that `content_type` matches the actual type of
-        content.
-        """
-        # Validate content_type
-        if self.content_type not in PYTHON_TYPES_FOR_CONTENT:
-            raise ValueError(
-                f"Content type unknown: '{self.content_type}'. "
-                f"Choose among: {', '.join(PYTHON_TYPES_FOR_CONTENT.keys())}"
-            )
-        if not isinstance(self.content, PYTHON_TYPES_FOR_CONTENT[self.content_type]):
-            raise ValueError(
-                f"The type of content ({type(self.content)}) does not match the "
-                f"content type: '{self.content_type}' expects '{PYTHON_TYPES_FOR_CONTENT[self.content_type]}'."
-            )
+        Generate the ID based on the init parameters.
+        """
         # Validate metadata
         for key in self.metadata:
             if key in [field.name for field in fields(self)]:
                 raise ValueError(f"Cannot name metadata fields as top-level document fields, like '{key}'.")
 
-        # Check if id_hash_keys are all present in the meta
-        for key in self.id_hash_keys:
-            if key not in self.metadata:
-                raise ValueError(
-                    f"'{key}' must be present in the metadata of the Document if you want to use it to generate the ID."
-                )
-        # Generate the ID
-        hashed_content = _create_id(
-            classname=self.__class__.__name__,
-            content=str(self.content),
-            metadata=self.metadata,
-            id_hash_keys=self.id_hash_keys,
-        )
-
         # Note: we need to set the id this way because the dataclass is frozen. See the docstring.
+        hashed_content = self._create_id()
         object.__setattr__(self, "id", hashed_content)
 
+    def _create_id(self):
+        """
+        Creates a hash of the content given that acts as the document's ID.
```

Comment:
```suggestion
        Creates a hash of the given content that acts as the document's ID.
```

The word order seems better like this, but the meaning is still correct, right? :)